### PR TITLE
Remove deprecated v1.10 options

### PR DESCRIPTION
**Pull request by jibi**
Remove deprecated v1.10 options:
* `k8s-watcher-queue-size`
* `blacklist-conflicting-routes`
* `device`
* `crd-wait-timeout` (operator)

I've requested the review of André, Deepesh, Martynas and Chris as you have marked these options as deprecated as so should have the most context on them.

### CODE REVIEW

File: Documentation/operations/upgrade.rst [hunk 1]
```diff
@@ -820,6 +817,12 @@ Removed options
   option now defaults to ``cluster-pool``.
 * ``--tofqdns-enable-poller``, ``--tofqdns-enable-poller-events``: These option
   were deprecated in Cilium 1.8 and are now removed
+* ``k8s-watcher-queue-size``: this option does not have any effect since 1.8 and
+  is now removed.
+* ``blacklist-conflicting-routes``: this option does not have any effect since
+  1.9 and is now removed.
+* ``device``: this option was deprecated in 1.9 in favor of ``devices`` and is
+  now removed.
```
Comment (aanm) [inline Documentation/operations/upgrade.rst]: This should be under lines 330-350

File: Documentation/operations/upgrade.rst [hunk 2]
```diff
@@ -805,9 +805,6 @@ Deprecated Metrics/Labels
 Deprecated options
 ~~~~~~~~~~~~~~~~~~
 
-* ``k8s-watcher-queue-size``: This option does not have any effect since 1.8
-  and is planned to be removed in 1.10.
-
```
Comment (aanm) [inline Documentation/operations/upgrade.rst]: This shouldn't be removed

File: daemon/cmd/daemon_main.go [hunk 3]
```diff
@@ -474,9 +465,6 @@ func init() {
 	option.BindEnv(option.K8sForceJSONPatch)
 	flags.MarkHidden(option.K8sForceJSONPatch)
 
-	flags.String(option.K8sWatcherEndpointSelector, defaults.K8sWatcherEndpointSelector, "K8s endpoint watcher will watch for these k8s endpoints")
-	option.BindEnv(option.K8sWatcherEndpointSelector)
```
Comment (aanm) [inline daemon/cmd/daemon_main.go]: This shouldn't be removed

File: pkg/k8s/watchers/endpoint.go [hunk 4]
```diff
@@ -19,19 +19,16 @@ import (
...
 	v1meta "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apimachinery/pkg/fields"
 	"k8s.io/apimachinery/pkg/util/wait"
 	"k8s.io/client-go/kubernetes"
 	"k8s.io/client-go/tools/cache"
 )
 
 func (k *K8sWatcher) endpointsInit(k8sClient kubernetes.Interface, swgEps *lock.StoppableWaitGroup, optsModifier func(*v1meta.ListOptions)) {
 	epOptsModifier := func(options *v1meta.ListOptions) {
-		options.FieldSelector = fields.ParseSelectorOrDie(option.Config.K8sWatcherEndpointSelector).String()
```
Comment (aanm) [inline pkg/k8s/watchers/endpoint.go]: This shouldn't be removed

File: pkg/option/config.go [hunk 5]
```diff
@@ -192,10 +189,6 @@ const (
 	// status in kube-apiserver.
 	K8sForceJSONPatch = "k8s-force-json-patch"
 
-	// K8sWatcherEndpointSelector specifies the k8s endpoints that Cilium
-	// should watch for.
-	K8sWatcherEndpointSelector = "k8s-watcher-endpoint-selector"
-
```
Comment (aanm) [inline pkg/option/config.go]: This shouldn't be removed

File: pkg/option/config.go [hunk 6]
```diff
@@ -939,7 +928,6 @@ var HelpFlagSections = []FlagsSection{
 			K8sRequireIPv4PodCIDRName,
 			K8sRequireIPv6PodCIDRName,
 			K8sSyncTimeoutName,
-			K8sWatcherEndpointSelector,
```
Comment (aanm) [inline pkg/option/config.go:942]: This shouldn't be removed

File: pkg/option/config.go [hunk 7]
```diff
@@ -2543,7 +2529,6 @@ func (c *DaemonConfig) Populate() {
 	c.K8sForceJSONPatch = viper.GetBool(K8sForceJSONPatch)
 	c.K8sEventHandover = viper.GetBool(K8sEventHandover)
 	c.K8sSyncTimeout = viper.GetDuration(K8sSyncTimeoutName)
-	c.K8sWatcherEndpointSelector = viper.GetString(K8sWatcherEndpointSelector)
```
Comment (aanm) [inline pkg/option/config.go]: This shouldn't be removed

**Review by aanm — CHANGES_REQUESTED** [review]

**Review by tklauser — COMMENTED** [review]
There is also the `crd-wait-timeout` in the operator which was deprecated in 325547f6ef3a. This one could be removed as part of this PR as well.

File: daemon/cmd/daemon_main.go (hunk 3 above)
Comment (jibi) [reply daemon/cmd/daemon_main.go]: ENOCOFFEE :facepalm:

**Review by fristonio — APPROVED** [review]
LGTM, Thanks :rocket:

**Review by qmonnet — APPROVED** [review]
Looks good, thank you!

**Review by tklauser — APPROVED** [review]
Thanks :rocket:

**Comment by tklauser** [comment]
test-me-please

**Review by brb — APPROVED** [review]
LGTM, thanks!

**Comment by jibi** [comment]
Looks like `K8s-1.19-kernel-4.9` is failing due to a known flake: https://github.com/cilium/cilium/issues/13071
I'll restart the test

**Comment by jibi** [comment]
retest-4.9

**Comment by jibi** [comment]
retest-net-next

**Review by aanm — APPROVED** [review]

**Review by christarazi — APPROVED** [review]
LGTM

**Pull request merged**